tests/test_atom_mapping.py

```diff
@@ -1520,3 +1520,44 @@ def test_enforce_chirally_valid_dummy_groups(seed):
     assert diagnostics_1.num_cores < diagnostics_1.total_leaves_visited
     for core in cores_1:
         verify_chiral_validity_of_core(mol_a, mol_b, core, ff)
+
+
+def test_ring_forming_breaking_count(hif2a_ligands):
+    mol_a = Chem.AddHs(Chem.MolFromSmiles("C1=CC=CC=C1"))  # benzene
+    set_mol_name(mol_a, "benzene")
+    mol_b = Chem.AddHs(Chem.MolFromSmiles("CC1=CC=CC=C1"))  # benzene with a methyl
+    set_mol_name(mol_b, "methyalated")
+    AllChem.EmbedMolecule(mol_a, randomSeed=2024)
+    AllChem.EmbedMolecule(mol_b, randomSeed=2024)
+    AllChem.AlignMol(mol_b, mol_a, atomMap=[(0, 0), (4, 3)])
+
+    # Construct core over an incomplete ring, resulting in a ring forming in both endstates
+    core = np.array(
+        [
+            [3, 3],
+            [4, 4],
+            [5, 5],
+        ]
+    )
+    assert atom_mapping.ring_forming_breaking_count(mol_a, mol_b, core) == 2
```

Comment:
Question: is it possible to determine the number of rings formed and broken separately? (e.g. could this return a tuple `(# rings formed, # rings broken)`)

Reply:
Mostly just wondering, don't have a concrete use case yet. It does seem nice to provide the additional information if it's an easy change (could imagine this being useful for "directional" estimates like SMC at some point).

Reply:
Added in https://github.com/proteneer/timemachine/pull/1363/commits/46e46f7f65c77ed68833693709e73b922c0756d8